### PR TITLE
Add build script for Improv at ANL LCRC

### PR DESCRIPTION
**Pull request by jtkrogel**
## Proposed changes

Enable builds on Improv at ANL's LCRC

## What type(s) of changes does this code introduce?

- Documentation or build script changes

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

Improv

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Review by prckent — COMMENTED** [review]
Is it not possible to `module load boost` etc. on this machine and avoid the extensive cmake line? Are they not setting the correct environment variables, making this extensive by-hand configuration necessary?

File: config/build_lcrc_improv.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+module purge
+module load cmake/3.27.4
+module load gcc/13.2.0
+module load openmpi/5.0.2-gcc-13.2.0
+module load amdlibflame/4.2-gcc-13.2.0
+module list
+
+
+CMAKE_FLAGS="-DENABLE_PPCONVERT=0 \
+             -DLAPACK_LIBRARIES=/gpfs/fs1/soft/improv/software/spack-built/linux-rhel8-zen3/gcc-13.2.0/openblas-0.3.26-blohgyt/lib/libopenblas.so \
```
Comment (ye-luo) [inline config/build_lcrc_improv.sh:30]: Loaded libflame but forcing openblas as the LAPACK?
Technically, amdblis is the BLAS and amdlibflame seems to be the LAPACK.
If you insist using openblas. Could you unload blis and flame?

**Comment by jtkrogel** [comment]
They do not have several of our needed dependencies available via module yet.

LCRC manages their system w/ Spack.  The admins quickly built a module for QMCPACK via Spack and it was easiest for now just to use paths to the dependencies Spack built along the way.  

I will continue to encourage them to make the dependencies available via modules.

File: config/build_lcrc_improv.sh (hunk 1 above)
Comment (jtkrogel) [reply config/build_lcrc_improv.sh]: Changed

File: config/build_lcrc_improv.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	     -DFFTW_LIBRARIES=/gpfs/fs1/soft/improv/software/spack-built/linux-rhel8-zen3/gcc-13.2.0/fftw-3.3.10-x5237xr/lib/libfftw3.so \
+	     -DFFTW_INCLUDE_DIR=/gpfs/fs1/soft/improv/software/spack-built/linux-rhel8-zen3/gcc-13.2.0/fftw-3.3.10-x5237xr/include \
+	     -DLIBXML2_INCLUDE_DIR=/gpfs/fs1/soft/improv/software/spack-built/linux-rhel8-zen3/gcc-13.2.0/libxml2-2.10.3-xkoaaap/include/libxml2 \
+	     -DLIBXML2_LIBRARY=/gpfs/fs1/soft/improv/software/spack-built/linux-rhel8-zen3/gcc-13.2.0/libxml2-2.10.3-xkoaaap/lib/libxml2.so \
```
[the remark this answers is not in the document]
Comment (jtkrogel) [reply config/build_lcrc_improv.sh]: Changed

**Comment by jtkrogel** [comment]
Comments addressed.

File: config/build_lcrc_improv.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+module load cmake/3.27.4
+module load gcc/13.2.0
+module load openmpi/5.0.2-gcc-13.2.0
+module list
+
+SPACK_ROOT=/gpfs/fs1/soft/improv/software/spack-built/linux-rhel8-zen3/gcc-13.2.0
+export CMAKE_PREFIX_PATH=$SPACK_ROOT/fftw-3.3.10-x5237xr:$CMAKE_PREFIX_PATH
+export CMAKE_PREFIX_PATH=$SPACK_ROOT/libxml2-2.10.3-xkoaaap:$CMAKE_PREFIX_PATH
+export CMAKE_PREFIX_PATH=$SPACK_ROOT/hdf5-1.14.3-6qo7t6e:$CMAKE_PREFIX_PATH
+export CMAKE_PREFIX_PATH=$SPACK_ROOT/openblas-0.3.26-blohgyt/lib:$CMAKE_PREFIX_PATH
```
Comment (ye-luo) [inline config/build_lcrc_improv.sh:29]: Probably `lib` is not needed.

**Review by ye-luo — COMMENTED** [review]
My last request. Please rename the file to
`build_anl_lcrc_improv.sh`

File: config/build_lcrc_improv.sh (hunk 3 above)
Comment (jtkrogel) [reply config/build_lcrc_improv.sh]: Correct

**Comment by jtkrogel** [comment]
The file has been renamed.  No issues remaining.

**Review by prckent — APPROVED** [review]
LGTM. All previous comments addressed

**Comment by prckent** [comment]
Test this please

**Pull request merged**